lib/rust/profiler/data/src/aggregate.rs

```diff
@@ -0,0 +1,87 @@
+//! Supports aggregating interval data by profiler to analyze total time spent, abstracting away
+//! *when* intervals occurred.
+
+use std::collections;
+use std::rc;
+
+/// Aggregate time spent in different functions.
+#[derive(Default, Debug)]
+pub struct Aggregator {
+    stack: Vec<std::rc::Rc<String>>,
+    root:  Frame,
+}
+
+impl Aggregator {
+    /// Add data from a profile to the tree.
+    pub fn visit_profile<Metadata>(&mut self, profile: &crate::Profile<Metadata>) {
```

Comment:
I'm confused why `visit_profile` is "addicting data from a profile to the tree". The name suggests to me that this is a standard visitor impl, but it looks it does something else. Do you think we could improve the naming here and in associated functions?

Reply:
Renamed the functions that apply a profile to the aggregated data `add_profile`.